packages/lib/gpc/src/gpcCompile.ts

```diff
@@ -488,6 +495,66 @@ function compileProofNumericValues(
   };
 }
 
+export function compileCommonEntryInequalities<
+  CompilerEntryInfo extends { entryConfig: GPCProofEntryConfig }
+>(
+  boundsCheckConfig: GPCProofBoundsCheckConfig,
+  entryMap: Map<PODEntryIdentifier, CompilerEntryInfo>,
+  paramEntryInequalities: number
+): {
+  entryInequalityValueIndex: CircuitSignal[];
+  entryInequalityOtherValueIndex: CircuitSignal[];
+  entryInequalityIsLessThan: CircuitSignal;
+} {
+  const numericValueIndex: Record<PODEntryIdentifier, bigint> =
```

Comment:
This relying on the sort order of an object and indirectly deriving the index feels fragile to me.
This PR is the point where the order of numeric value modules is becoming important, because it's becoming an input to another module.  That might make it a good time to restructure the data in boundsCheckConfig.  Call it numericValueConfig, or call it a NumericValueInfo map by analogy to the ObjInfo and EntryInfo maps.  Then have it explicitly contain the index as a field, much like objInfo map contains the obj index as a field.  Conceptually each of those maps is similar in purpose, and they build on each other in the same way.